src/Servers/Kestrel/Transport.Sockets/ref/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.netcoreapp.cs

```diff
@@ -26,3 +26,13 @@ public partial class SocketTransportOptions
         public bool NoDelay { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
     }
 }
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Client
+{
+    public partial class SocketConnectionFactory : Microsoft.AspNetCore.Connections.IConnectionFactory, System.IAsyncDisposable
```

Comment:
I'm fine with making this public in 5.0 for now, but is there any other option besides this? Does InternalsVisibleTo not work here?

Reply:
InternalsVisibleTo isn't really an option when we need to consume this across several projects, it doesn't scale. Stephen said they wanted to make this public soon anyways.

Reply:
I know we wanted to improve this API to take in a propertybag/featurecollection. That may be something we want to consider before making this public?

Reply:
We'll run it by API review Monday and see. If it's just additions then there's no need to block this PR.

Reply:
Scratch that, Servers aren't subject to API review yet. I'll check the current API with @halter73 and file any follow up issues. We have the entire release to work out the API surface and I want to get this in to unblock HttpSys feature development.

Reply:
Given the namespace soup here (that it's deep within `.Kestrel.Transport.Sockets.Client`) I feel OK making this public since it won't collide with a more "real" client factory.